File: frontend/src/api/reviews.ts
import type { ApiResponse } from './request';
import { request } from './request';

import type { AgentExecutionLog, ReviewTask } from '../types';

export function startFullReview(contractId: number | string) {
  return request.post<ApiResponse<{ task: ReviewTask; profile: Record<string, unknown> }>>(
    `/api/v1/contracts/${contractId}/review`,
  );
}

export function startProfileReview(contractId: number | string) {
  return request.post<ApiResponse<{ task: ReviewTask; profile: Record<string, unknown> }>>(
    `/api/v1/contracts/${contractId}/review/profile`,
  );
}

export function getReviewTask(taskId: number | string) {
  return request.get<ApiResponse<ReviewTask>>(`/api/v1/review-tasks/${taskId}`);
}

export function getReviewProgress(taskId: number | string) {
  return request.get<ApiResponse<{ task: ReviewTask; agent_logs: AgentExecutionLog[] }>>(
    `/api/v1/review-tasks/${taskId}/progress`,
  );
}

export function getAgentLogs(taskId: number | string) {
  return request.get<ApiResponse<AgentExecutionLog[]>>(`/api/v1/review-tasks/${taskId}/agent-logs`);
}

export function getProfileResult(taskId: number | string) {
  return request.get<ApiResponse<Record<string, unknown>>>(`/api/v1/review-tasks/${taskId}/profile`);
}

export function getReviewClauses(taskId: number | string) {
  return request.get<ApiResponse<Record<string, unknown>[]>>(`/api/v1/review-tasks/${taskId}/clauses`);
}

export function getReviewRisks(taskId: number | string) {
  return request.get<ApiResponse<Record<string, unknown>[]>>(`/api/v1/review-tasks/${taskId}/risks`);
}

export function getReviewSuggestions(taskId: number | string) {
  return request.get<ApiResponse<Record<string, unknown>[]>>(`/api/v1/review-tasks/${taskId}/suggestions`);
}

export function getReviewReport(taskId: number | string) {
  return request.get<ApiResponse<Record<string, unknown>>>(`/api/v1/reports/${taskId}`);
}

export function getLatestReview(contractId: number | string) {
  return request.get<ApiResponse<ReviewTask | null>>(`/api/v1/contracts/${contractId}/review/latest`);
}

File: backend/app/api/v1/reviews.py
from fastapi import APIRouter

from app.core.response import success

router = APIRouter()


@router.post("/contracts/{contract_id}/review")
def start_review_placeholder(contract_id: int):
    return success({"contract_id": contract_id, "implemented": False}, "审查任务接口骨架已创建")


@router.get("/review-tasks/{task_id}")
def review_task_placeholder(task_id: int):
    return success({"task_id": task_id, "implemented": False})


@router.get("/review-tasks/{task_id}/progress")
def review_progress_placeholder(task_id: int):
    return success({"task_id": task_id, "steps": [], "implemented": False})


@router.get("/review-tasks/{task_id}/agent-logs")
def agent_logs_placeholder(task_id: int):
    return success({"task_id": task_id, "items": [], "implemented": False})


@router.get("/review-tasks/{task_id}/profile")
def profile_placeholder(task_id: int):
    return success({"task_id": task_id, "profile": None, "implemented": False})


@router.get("/review-tasks/{task_id}/clauses")
def clauses_placeholder(task_id: int):
    return success({"task_id": task_id, "items": [], "implemented": False})


@router.get("/review-tasks/{task_id}/risks")
def risks_placeholder(task_id: int):
    return success({"task_id": task_id, "items": [], "implemented": False})


@router.get("/review-tasks/{task_id}/suggestions")
def suggestions_placeholder(task_id: int):
    return success({"task_id": task_id, "items": [], "implemented": False})
